### ISSUE TITLE
problems about the data preprocessing

### ISSUE THREAD

**Issue by lyy-zz** [shicai/DenseNet-Caffe#30]
When i use the pycaffe to prediction, according to readme  subtract mean [103.94,116.78,123.68],but the result is  bad.I also suspect that it is a label problem,so i follow the tips for another question,check the [fb.resnet.torch/imagenet.lua at master · facebook/fb.resnet.torch](https://github.com/facebook/fb.resnet.torch/blob/master/pretrained/imagenet.lua) have no problems.code as follows:
```
    transformer=caffe.io.Transformer({'data':net.blobs['data'].data.shape})
    transformer.set_transpose('data',(2,0,1))
    transformer.set_mean('data',np.array([103.94,116.78,123.68]))
    transformer.set_raw_scale('data',225)
    transformer.set_channel_swap('data',(2,1,0))
```
So i tried to comment some preprocesss setting,when i comment the`transformer.set_mean` and `transformer.set_raw_scale`,i found the prediction is very good.
I am very confused,train and test need to be consistent in general ??
I want to ask where the problem might be. 
Looking forward to anyone’s reply, thank you!

**Comment by shicai** [comment]
don't comment `set_mean` and `set_raw_scale`.
please add anther line of code `transformer.set_input_scale('data', 0.017)`.
it should be ok.

**Comment by lyy-zz** [comment]
Yes,you are right, is ok now ! sorry,this is the first time i use this param,i will learn more about it.
Thank you very much for your reply!
